index.js

```diff
@@ -34,12 +34,14 @@ var nightwatchPlugin = function(options) {
       child.kill();
     }
 
-    if (stream) {
+    if(options.abortOnFailure === true && stream) {
       if (code) {
         stream.emit('error', new PluginError(PLUGIN_NAME, 'nightwatch exited with code ' + code));
       } else {
         stream.emit('end');
       }
+    } else {
+      stream.emit('end');
```

Comment:
I think when `stream` is `undefined`, calling `stream.emit` fails.

How about the following code:

```javascript
if (stream) {
  if (options.abortOnFailure) {
     stream.emit('end');
  } else if (code) {
     stream.emit('error', new PluginError(PLUGIN_NAME, 'nightwatch exited with code ' + code));
  } else {
     stream.emit('end');
  } 
}
```

Reply:
That's right, I did the reverse.
False by default, it issues End of Error instead.

Reply:
Sorry, the above code I wrote is wrong...
The below code is a correct example:

```ruby
if (stream) {
  if (code && options.abortOnFailure) {
     stream.emit('error', new PluginError(PLUGIN_NAME, 'nightwatch exited with code ' + code));
  } else {
     stream.emit('end');
  } 
}
```

> I think when stream is undefined, calling stream.emit fails.

I care about that if `options.abortOnFailure` is `true` and `stream` is `undefined` at the line 37 condition, `Uncaught TypeError: Cannot read property 'emit' of undefined` occurs at the line 44.
https://github.com/tatsuyafw/gulp-nightwatch/blob/dddef7032f8119e680244c84b458ab22b842db32/index.js#L37-L45